src/pages/LearnPage/LearnPage.tsx

```diff
@@ -100,124 +164,136 @@ function LearnPage() {
-                    height: "auto",
-                    justifyContent: "flex-end",
-                  }}
+                  className="lesson-subtitle"
+                  style={{ color: theme.palette.primary.dark }}
                 >
-                  <div
-                    className="lesson-subtitle"
-                    style={{ color: theme.palette.primary.dark }}
-                  >
-                    <span style={{ color: theme.palette.primary.main }}>
-                      Author:{" "}
-                    </span>
-                    {sampleLesson[0].author}
-                  </div>
-                  <div
-                    className="lesson-subtitle"
-                    style={{ color: theme.palette.primary.dark }}
-                  >
-                    <span style={{ color: theme.palette.primary.main }}>
-                      Last Updated:{" "}
-                    </span>
-                    {sampleLesson[0].date}
-                  </div>
+                  <span style={{ color: theme.palette.primary.main }}>
+                    Author:{" "}
+                  </span>
+                  {currentLesson.author}
                 </div>
-              </LessonHeader>
-              {sampleLesson.slice(1).map((block: any, i) => {
-                if (block.type === "text") {
-                  console.log(JSON.parse(JSON.stringify(block.content)));
+                <div
+                  className="lesson-subtitle"
+                  style={{ color: theme.palette.primary.dark }}
+                ></div>
+              </div>
+            </LessonHeader>
+            {currentLesson.content?.map((block: any, i) => {
+              if (!block || !block.type || !block.content) {
+                return null;
+              }
+              if (block.type === "text") {
+                console.log(JSON.parse(JSON.stringify(block.content)));
 
-                  return (
-                    <div key={i} style={{ width: "100%" }}>
-                      <Markup
-                        transform={transform}
-                        className="inter"
-                        content={JSON.parse(JSON.stringify(block.content.html))}
-                      />
-                    </div>
-                  );
-                } else if (block.type === "image") {
-                  return (
-                    <ImageBlock
-                      key={i}
-                      src={block.content.sourcePath}
-                      caption={block.content.caption}
-                      height={block.content.height}
-                      width={block.content.width}
-                      alignment={block.content.alignment}
+                return (
+                  <div key={i} style={{ width: "100%" }}>
+                    <Markup
+                      transform={transform}
+                      className="inter"
+                      content={JSON.parse(JSON.stringify(block.content.html))}
```

Comment:
json.parse and json stringify can throw can we have a try catch wrapped around this?